java-components/build-request-processor/src/test/java/com/redhat/hacbs/container/analyser/deploy/DeployContaminateTest.java

```diff
@@ -0,0 +1,161 @@
+package com.redhat.hacbs.container.analyser.deploy;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.logging.LogRecord;
+
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.inject.Inject;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.redhat.hacbs.container.results.ResultsUpdater;
+import com.redhat.hacbs.resources.util.HashUtil;
+
+import io.quarkus.test.LogCollectingTestResource;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.ResourceArg;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@QuarkusTestResource(value = LogCollectingTestResource.class, restrictToAnnotatedClass = true, initArgs = @ResourceArg(name = LogCollectingTestResource.LEVEL, value = "FINE"))
```

Comment:
I looked at https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Test.20Logging.3F which led to a few options but using  https://github.com/quarkusio/quarkus/commit/7c81682e59411546a6d381f843fe4644a5bbe81d seemed simplest

Reply:
To be honest I don't really like tests like this, they are really fragile and tied to the implementation. 

I would create two repos at https://github.com/orgs/jvm-build-service-test-data/repositories , one called 'unbuildable' or something like that, and one where the test jar is contaminated by unbuildable. I think you should have access, both to be able to create the repos and release to central.

Then add the gav of the contaiminated one to the minikube test matrix. 

Over time I want to build up https://github.com/orgs/jvm-build-service-test-data to contain all sorts of different problems we have solved, but that only have one or two classes so they build super quick.